### PR TITLE
Adding tests for React.Fragment, Children as Array and Children as function

### PR DESCRIPTION
**Pull request by antonioru**
Adding few tests for React.Fragment and Children as Array conversion

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -1958,6 +1958,133 @@ test('inline registration of component args - arrow function property lookup', a
...
+
+test('React.Fragment', async ({ is }) => {
+  const src = [
+    `const React = require("react")`,
+    `const Cmp = () => <React.Fragment><div>Foo</div></React.Fragment>`,
+    `module.exports = Cmp`
+  ].join('\n')
+
+  const esx = [
+    `const esx = require('esx')()`,
```
Comment (davidmarkclements) [inline test/index.test.js:1970]: Not a big deal but avoid mixing “ and ‘ (stick with ‘)

Comment (antonioru) [reply test/index.test.js:1970]: Ops... sorry about that :)

File: test/index.test.js [hunk 2]
```diff
@@ -1958,32 +1958,187 @@ test('inline registration of component args - arrow function property lookup', a
...
+    'const Repeater = (props) => esx `<p>${props.children("foo")}</p>`',
+    `esx.register({ Repeater });`,
+    'const App = () => esx `<Repeater>${(str) => esx `<h1>${str}</h1>`}</Repeater>`;',
+    `module.exports = App`,
+  ].join('\n');
+
+  is(convert(src), esx);
+});
+
+test('fuckadoo', async ({ is }) => {
```
Comment (davidmarkclements) [inline test/index.test.js:2088]: we need to rename this hahah 

this was a case where it was failing, is it now passing? if it's passing for now let's call it 'complex case' 

if it is failing, let's whittle down the code to isolate the failure exactly, and then name the test accordingly

Comment (antonioru) [reply test/index.test.js]: @davidmarkclements got it, I renamed the test 

Yes, the tests are all passing even tho I would double check this precise case

**Comment by davidmarkclements** [comment]
great work!

**Pull request merged**